CMakeLists.txt

```diff
@@ -92,6 +92,47 @@ if ("${GOOGLE_CLOUD_CPP_ENABLE_CCACHE}")
     endif ()
 endif ()
 
+# The default source for dependencies.
+set(GOOGLE_CLOUD_CPP_DEPENDENCY_PROVIDER "external" CACHE STRING "[===[
+How to find the dependencies for google-cloud-cpp.
+
+The Google Cloud C++ client libraries have several external dependencies,
+notably gRPC, libcurl, OpenSSL, and a few others.
+
+This configuration option controls how are these dependencies are found. It can
+be set to:
+
+* external: use CMake's `ExternalProject` to download and compile the
+  dependencies. This is the recommended configuration for development, as it
+  isolates your build from the system configuration.
+
+* package: use CMake's `find_package()` to find the dependencies. This is the
+  recommended configuration for compiling and installing google-cloud-cpp.
+
+* vcpkg: use github.com/Microsoft/vcpkg to find the dependencies. vcpkg
+  largely behaves as `package`, but we have a few bugs in our CMake
+  configuration that need to be cleaned up. This option will be changed to
+  behave exactly as `package` once we fix these problems.
+
+* pkg-config: use this if the dependencies have been installed, but without
+  CMake support. Sometimes using pkg-config(1) can work as an alternative way
+  to find the dependency. WARNING: this is poorly tested, use at your own risk.
```

Comment:
Can we phrase this a bit more nicely than "poorly tested"?  This comes across like we should have tested it better but didn't... maybe go with "largely untested in favor of using `external` or `package` instead", or something that indicates that we prefer, and more heavily test, those modes?

Reply:
Agree, let me know what you think of the new version.